src/cascade/core/context.py

```diff
@@ -48,9 +44,12 @@ def __init__(self):
 
 
 class _Integrand:
```

Comment:
I am taking a moment to review the entire project structure, and code.
At first glance, I am noticing that context.py imports Rate: 
from cascade.model.rates import Rate
and defines Rates, as well as Integrand and Integrands.

Reply:
Yeah, I need to clean this stuff up. There's debris from several changed decisions and mistakes. Drew was pointing that out too. Not in this PR but I'm going to do that in the next week or two. I'll get you and Drew to go over it with me too. There are several technical debt issues like this we need to address now that the structure of the problem is more clear. Being more organized about our use of exceptions is another one.